src/custom/hooks/useRefetchPriceCallback.ts

```diff
@@ -49,34 +55,49 @@ class PriceQuoteError extends Error {
 
 type PriceSource = 'gnosis-protocol' | 'paraswap'
 type PriceInformationWithSource = PriceInformation & { source: PriceSource; data?: any }
+type PromiseRejectedResultWithSource = PromiseRejectedResult & { source: PriceSource }
 
 /**
  *
  * @returns The best price among GPv2 API or Paraswap one
  */
 async function _getBestPriceQuote(params: PriceQuoteParams): Promise<PriceInformation> {
   // Get price from all API: Gpv2 and Paraswap
-  const pricePromise = getPriceQuote(params) // TODO: Add timeout (another PR)
-  const paraSwapPricePromise = getPriceQuoteParaswap(params) // TODO: Add timeout (another PR)
+  const pricePromise = withTimeout(getPriceQuote(params), PRICE_API_TIMEOUT_MS, 'GPv2: Get Price API')
+  const paraSwapPricePromise = withTimeout(
+    getPriceQuoteParaswap(params),
+    PRICE_API_TIMEOUT_MS,
+    'Paraswap: Get Price API'
+  )
 
   // Get results from API queries
   const [priceResult, paraSwapPriceResult] = await Promise.allSettled([pricePromise, paraSwapPricePromise])
-  const price = getPromiseFulfilledValue(priceResult, null)
-  const paraswapPriceRaw = getPromiseFulfilledValue(paraSwapPriceResult, null)
 
   // Prepare an array with all successful estimations
   const priceQuotes: Array<PriceInformationWithSource> = []
-  if (price) {
-    priceQuotes.push({ ...price, source: 'gnosis-protocol' })
+  const errorsGetPrice: Array<PromiseRejectedResultWithSource> = []
+
+  if (isPromiseFulfilled(priceResult)) {
+    priceQuotes.push({ ...priceResult.value, source: 'gnosis-protocol' })
+  } else {
+    errorsGetPrice.push({ ...priceResult, source: 'gnosis-protocol' })
   }
-  if (paraswapPriceRaw) {
-    const paraswapPrice = toPriceInformation(paraswapPriceRaw)
-    paraswapPrice && priceQuotes.push({ ...paraswapPrice, source: 'paraswap', data: paraswapPriceRaw })
+
+  if (isPromiseFulfilled(paraSwapPriceResult)) {
+    const paraswapPrice = toPriceInformation(paraSwapPriceResult.value)
+    paraswapPrice && priceQuotes.push({ ...paraswapPrice, source: 'paraswap', data: paraSwapPriceResult.value })
+  } else {
+    errorsGetPrice.push({ ...paraSwapPriceResult, source: 'paraswap' })
   }
 
-  console.log('[hooks::useRefetchPriceCallback] _getBestPriceQuote', priceQuotes)
+  if (errorsGetPrice.length > 0) {
+    const sourceNames = errorsGetPrice.map(e => e.source).join(', ')
+    console.error('[hooks::useRefetchPriceCallback] Some API failed or timed out: ' + sourceNames, errorsGetPrice)
+  }
 
   if (priceQuotes.length > 0) {
+    const sourceNames = priceQuotes.map(p => p.source).join(', ')
+    console.log('[hooks::useRefetchPriceCallback] Get best price succeeded for ' + sourceNames, priceQuotes)
```

Comment:
You could consider listing all our price end-points with `console.table()` to provide a nice visual overview of all end-points, prices and the last column 'WINNING PRICE' with a ✅.

Reply:
good idea, will do in new PR

Reply:
https://github.com/gnosis/cowswap/issues/841